src/main/java/com/google/ddexmeadparser/DdexMeadParser.java

```diff
@@ -0,0 +1,258 @@
+package com.google.ddexmeadparser;
+
+import com.google.ddexmeadparser.experimental.SchemaSetMerger;
+import com.google.protobuf.Message;
+
+import org.apache.commons.cli.*;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.stream.StreamSource;
+import java.io.*;
+import java.util.Set;
+
+import com.google.common.flogger.FluentLogger;
+
+/** The type Ddex mead parser. */
+public class DdexMeadParser {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private final DdexMeadParserOptions runtimeOptions;
+
+  /**
+   * The entry point of application.
+   *
+   * @param args the input arguments
+   */
+  public static void main(String[] args) {
+    try {
+      DdexMeadParser ddexMeadParser = new DdexMeadParser(args);
+      ddexMeadParser.exec();
+    } catch (InvalidOptionsException | MeadConversionException | SchemaConversionException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Instantiates a new Ddex mead parser. Serves as entry point for command line usage
+   *
+   * @param args the args
+   * @throws InvalidOptionsException the invalid options exception
+   */
+  public DdexMeadParser(String[] args) throws InvalidOptionsException {
+    Options commandOptions = buildCommandOptions();
+    try {
+      CommandLineParser commandParser = new DefaultParser();
+      CommandLine commandLineInput = commandParser.parse(commandOptions, args);
+      runtimeOptions = buildRuntimeOptions(commandLineInput);
+    } catch (ParseException e) {
+      showCommandUsage(commandOptions);
+      throw new InvalidOptionsException("Parse exception occurred: " + e.getMessage(), e);
+    } catch (InvalidOptionsException e) {
+      showCommandUsage(commandOptions);
+      throw e;
+    }
+  }
+
+  /**
+   * Exec.
+   *
+   * @throws MeadConversionException the mead conversion exception
+   * @throws SchemaConversionException the schema conversion exception
+   * @throws InvalidOptionsException the invalid options exception
+   */
+  public void exec()
```

Comment:
Please try to use the lowest visibility possible. If this method is only called by the main() in this class, it can be private.